packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -765,6 +765,16 @@ class _WidgetTextStyleMapper extends WidgetStateTextStyle {
 /// of their current material state and [resolve] the button style's
 /// material state properties when their value is needed.
 ///
+/// {@tool dartpad}
+/// This example shows how you can override the default text and icon
```

Comment:
Just noticed the "you", from the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-the-passive-voice-recommend-do-not-require-never-say-things-are-simple) it should be avoided.
Maybe something such as "This example shows how the default text and icon color (the "foreground color") of a [TextButton] can be overridden with a [WidgetStateProperty]."